go_lib/bassa.go

```diff
@@ -0,0 +1,738 @@
+		panic(errIncompleteParams)
+	}
+	endpoint := "/api/login"
+	apiURL := b.apiURL + endpoint
+
+	form := url.Values{}
+	form.Add("user_name", userName)
+	form.Add("password", password)
+
+	response, err := http.PostForm(apiURL, form)
+	if err != nil {
+		panic(err)
+	}
+	defer response.Body.Close()
+	b.token = response.Header["Token"][0]
+
+	respBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		fmt.Println(string(respBody))
+		panic(err)
+	}
+}
+
+// AddRegularUserRequest : Function to login as a user
+func (b *Bassa) AddRegularUserRequest(userName string, password string, email string) {
+	if userName == "" || password == "" || email == "" {
+		panic(errIncompleteParams)
+	}
+
+	err := validateFormat(email)
+	if err != nil {
+		panic(err)
+	}
+
+	endpoint := "/api/regularuser"
+	apiURL := b.apiURL + endpoint
+
+	requestBody, err := json.Marshal(map[string]string{
+		"user_name": userName,
+		"password":  password,
+		"email":     email})
+	if err != nil {
+		panic(err)
+	}
+	request, err := http.NewRequest("POST", apiURL, bytes.NewBuffer(requestBody))
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+
+	respBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		fmt.Println(string(respBody))
+		panic(err)
+	}
+}
+
+// AddUserRequest : Function to login as a user
+func (b *Bassa) AddUserRequest(userName string, password string, email string, authLevel int) {
+	if userName == "" || password == "" || email == "" {
+		panic(errIncompleteParams)
+	}
+
+	err := validateFormat(email)
+	if err != nil {
+		panic(err)
+	}
+
+	endpoint := "/api/user"
+	apiURL := b.apiURL + endpoint
+
+	requestBody := []byte(fmt.Sprintf("{user_name:\"%s\", password: \"%s\", email: \"%s\", auth: %d}", userName, password, email, authLevel))
+
+	request, err := http.NewRequest("POST", apiURL, bytes.NewBuffer(requestBody))
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	fmt.Println(string(out))
+}
+
+// RemoveUserRequest : Function to remove user
+func (b *Bassa) RemoveUserRequest(userName string) string {
+	if userName == "" {
+		panic(errIncompleteParams)
+	}
+
+	endpoint := "/api/user" + "/" + userName
+	apiURL := b.apiURL + endpoint
+
+	request, err := http.NewRequest("DELETE", apiURL, nil)
+	if err != nil {
+		panic(err)
+	}
+	request.Header.Set("token", b.token)
+	response, err := b.httpClient.Do(request)
+	if err != nil {
+		panic(err)
+	}
+
+	defer response.Body.Close()
+	var r interface{}
+	if err := json.NewDecoder(response.Body).Decode(&r); err != nil {
+		panic(err)
+	}
+	out, err := prettyjson.Marshal(r)
+	return string(out)
```

Comment:
what kind of information you want to return here? success or not? maybe have a different struct to have the proper data? and decode json using that?